### PR TITLE
move jwt from query string to headers

### PR DESCRIPTION
**Pull request by aleksei-okatiev**
Atlassian deprecated and removed JWT as a query param, now it should be in a header.
https://developer.atlassian.com/changelog/#CHANGE-995

### CODE REVIEW

File: lib/jira/jwt_client.rb [hunk 1]
```diff
@@ -14,6 +15,30 @@ def make_multipart_request(url, data, headers = {})
...
+          http_method.to_s,
+          site,
+          (Time.now - 60).to_i,
+          (Time.now + 86_400).to_i
+
+        JWT.encode claim, shared_secret
+      end
+    end
+
     class JwtUriBuilder
```
Comment (adam-mazur-gat) [inline lib/jira/jwt_client.rb:42]: Is it used somewhere outside of this file?
The only usage in this file is removed `JwtUriBuilder`

File: lib/jira/jwt_client.rb [hunk 2]
```diff
@@ -14,6 +15,30 @@ def make_multipart_request(url, data, headers = {})
       super(url, data, headers)
     end
 
+    class JwtBuilder
```
Comment (adam-mazur-gat) [inline lib/jira/jwt_client.rb:18]: If `build_jwt` is the only user of jwt building do we need a class for this? Can we just 

```ruby
 claim = Atlassian::Jwt.build_claims \
          issuer,
          request_url,
          http_method.to_s,
          site,
          (Time.now - 60).to_i,
          (Time.now + 86_400).to_i

        JWT.encode claim, shared_secret
```

in the `build_jwt`?

**Review by adam-mazur-gat — CHANGES_REQUESTED** [review]

File: lib/jira/jwt_client.rb [hunk 3]
```diff
@@ -4,8 +4,9 @@ module JIRA
   class JwtClient < HttpClient
     def make_request(http_method, url, body = '', headers = {})
       @http_method = http_method
+      @jwt = build_jwt(url)
 
-      super(http_method, url, body, headers)
+      super(http_method, url, body, headers.merge(jwt_header))
     end
 
     def make_multipart_request(url, data, headers = {})
```
Comment (gat-developer) [inline lib/jira/jwt_client.rb:12]: Should we do the same changes as for `make_request` here?

**Comment by bobbrodie** [comment]
Closing as dupe of #412

**Pull request closed without merging**